### PR TITLE
Added e2e tests for auditor read-only forms

### PR DESCRIPTION
**Pull request by mabashian**
First pass at adding some e2e tests that will confirm that fields appear as read-only in the UI for an auditor user.  In order to achieve this, I needed to create the following objects:

Inventories
Inventory Scripts
Notification Templates
Organizations
Projects
Teams
Templates
Users

The idea is that these objects can be used when testing other bits of expected behavior on those pages.  Currently, the test-auditor-read-only-forms test:
1) Logs in as the e2e admin
2) Creates all the necessary resources to perform the test(s) including the auditor user
3) Logs in as the auditor user and navigates to the forms in question and checks to see that all fields are disabled and save buttons are hidden.

Things left to do:
1) There's a race condition on the JT form that sometimes causes the test to fail. As a short-term workaround I added a pause after navigating to the JT form before checking the fields.  I want to look a little deeper in to why this is happening.  My hunch is it has something to do with spinner disappearing before all the data is loaded.
2) The test-credentials-read-only test can be removed as it's duplicated in the included auditor test.
3) ~~A new test needs to be added to make sure the save button is hidden when the auditor user is viewing the form.~~ see: https://github.com/ansible/awx/issues/260

### CODE REVIEW

**Comment by mabashian** [comment]
@jakemcdermott 

> There's a race condition on the JT form that sometimes causes the test to fail. As a short-term workaround I added a pause after navigating to the JT form before checking the fields. I want to look a little deeper in to why this is happening. My hunch is it has something to do with spinner disappearing before all the data is loaded.

I've taken a quick look at this issue and I'm positive that what's happening is that the spinner is disappearing before _all_ the data is loaded.  If we add logging to the Wait service you'll see that we call start at least two times and stop numerous other times.  In order to fix this properly we'd need to run through the JT edit controller logic and clean up these `Wait` calls so that it behaves as expected: `Wait('start');` called when data is initially being loaded.  `Wait('stop');` when _all_ the data is done being loaded.  I wanted to see what your thoughts were on whether the pause I've included in the JT portion of the test is a blocker for getting this merged and/or if we need to the `Wait` issues resolved before merging this.

**Comment by jakemcdermott** [comment]
@mabashian 
We shouldn't merge any hard-coded delays, so in this case we'd need to either update the controller or hold off on adding tests for this particular form for now.

File: awx/ui/client/test/e2e/tests/test-auditor-read-only-forms.js
Comment (jakemcdermott) [inline awx/ui/client/test/e2e/tests/test-auditor-read-only-forms.js]: We should pull this logic into a generic "checkAllFieldsDisabled" command for form sections. You could add that here: https://github.com/ansible/awx/blob/devel/awx/ui/client/test/e2e/objects/sections/createFormSection.js#L89

something like this
```
const checkAllFieldsDisabled = function() {
    let possibleInputs = [
        '.at-Input',
        '.Form-textInput',
        'select.Form-dropDown',
        'input[type="checkbox"]',
        'input[type="radio"]',
        '.ui-spinner-input',
        '.Form-textArea',
        '.ScheduleToggle-switch',
        '.Form-lookupButton'
    ];
    possibleInputs.forEach(function(selector) {
        // ...
    })
};

const createFormSection = function({ selector, labels, strategy }) {
    let options = generatorOptions[strategy || 'default'];

    let formSection = {
        selector,
        sections: {},
        elements: {},
        commands: [{
            checkAllFieldsDisabled
        ]}
    };
```

**Review by jakemcdermott — CHANGES_REQUESTED** [review]
LGTM overall - I think this is good to go after adding the `checkAllFieldsDisabled`. We don't necessarily need to fix the JT form here, though we would need to remove the JT test if we don't get to it here.

File: awx/ui/client/test/e2e/tests/test-auditor-read-only-forms.js
Comment (jakemcdermott) [inline awx/ui/client/test/e2e/tests/test-auditor-read-only-forms.js]: Since we're populating the 'created' field in one shot after doing the setup, we probably don't need to destructure the callback args here
```
...
},
 created => {
    store.created = created;
}
```

File: awx/ui/client/test/e2e/tests/test-auditor-read-only-forms.js
Comment (jakemcdermott) [inline awx/ui/client/test/e2e/tests/test-auditor-read-only-forms.js]: You should only need to call `useCss()` once here

**Comment by mabashian** [comment]
After more discussion we've decided to remove/disable the JT test(s) for now and wait until the JT edit controller is refactored a bit.  Having a reliable `Wait('start');`, `Wait('stop');` pattern will remove the need for hard coded pauses and ensure that we don't have intermittent race condition caused failures in our test runs.

**Review by jakemcdermott — APPROVED** [review]

File: awx/ui/client/test/e2e/objects/sections/createFormSection.js
Comment (jakemcdermott) [inline awx/ui/client/test/e2e/objects/sections/createFormSection.js:92]: @mabashian 

This LGTM

**Pull request merged**
